front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -41,6 +45,14 @@ class IntentMenu extends Component {
         this.setState({inputValue: event.target.value});
     }
 
+    handleStarChange() {
+        this.setState({starFilter: !this.state.starFilter});
+    }
+
+    handleFlagChange() {
+        this.setState({flagFilter: !this.state.flagFilter});
+    }
+
```

Comment:
Instead of handling this in local state, I think it makes more sense to handle this in redux. We can have redux variables that represent filtering by star/flag and then build those into our already existing filter functions

Reply:
I'm not sure which filter functions you mean? I will definitely change it to handle it in redux

Reply:
In components/helpers we have a general filter function. It would probably be best to re-use its logic (maybe making small changes) for this.

Reply:
I looked into the filter function and it's built for string searching. I don't think it would be possible to do it by an object attribute so I'll implement a separate filter function.

Reply:
We can modify the function to accept star and flag params and then filter by those as well. We should absolutely use it.